model/import/Repository/MetadataRepository.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoQtiItem\model\import\Repository;
+
+use oat\oatbox\service\ConfigurableService;
+use oat\generis\model\GenerisRdf;
+use core_kernel_classes_Class;
+use tao_helpers_form_elements_Htmlarea as HtmlArea;
+use tao_helpers_form_elements_Textarea as TextArea;
+use tao_helpers_form_elements_Textbox as TextBox;
+
+class MetadataRepository extends ConfigurableService
+{
+
+    private const TEXT_WIDGETS = [
+        TextBox::WIDGET_ID,
+        TextArea::WIDGET_ID,
+        HtmlArea::WIDGET_ID,
+    ];
+
+    /**
+     * Get metadata list
+     * @return array
+     */
+    public function getMetadataArray(string $uri): array
+    {
+        $metaDataArray      = array();
+        $class              = new core_kernel_classes_Class($uri);
+        $aliasProperty      = $class->getProperty(GenerisRdf::PROPERTY_ALIAS);
+        $classProperties    = $class->getProperties(true);
+        foreach ($classProperties as $property) {
+            $aliasName = (string)$property->getOnePropertyValue($aliasProperty);
+            if (!$property->getWidget()) {
+                continue;
+            }
+            $widgetUri = $property->getWidget()->getUri();
+            if ($aliasName && $this->isTextWidget($widgetUri)) {
+                $metaDataArray[] = "metadata_" . $property->getLabel() . "_" . $aliasName;
+            }
+        }
+        return $metaDataArray;
+    }
+
+    /**
+     * Check whether it is a text widget
+     * @return bool
+     */
+    private function isTextWidget($widgetUri): bool
+    {
+        return ($widgetUri)
+            ? in_array($widgetUri, self::TEXT_WIDGETS, true)
+            : false;
+    }
```

Comment:
Are you sure code above works as expected? As for me it may give false results, as can accept any argument and immediately return true

```suggestion
    private function isTextWidget(string $widgetUri): bool
    {
        return in_array($widgetUri, self::TEXT_WIDGETS, true);
    }
```

Reply:
Done